lib/hydra/derivatives/processors/active_encode.rb

```diff
@@ -5,18 +5,23 @@ class ActiveEncode < Processor
     def process
       encode = ::ActiveEncode::Base.create(source_path, directives)
 
-      # TODO: wait until the encode job succeeds then call output_file_service with the output url
-      # while(encode.reload.running?) do
-      #   p "Wait: #{Time.now}"
-      #   sleep 10
-      # end
+      # Wait until the encoding job is finished
+      # while(encode.reload.running?) { sleep 10 }
 
       raise_exception_if_encoding_failed(encode)
+      raise_exception_if_encoding_cancelled(encode)
+
+      # TODO: call output_file_service with the output url
     end
 
     def raise_exception_if_encoding_failed(encode)
       return unless encode.failed?
-      raise StandardError.new("Encoding failed: #{encode.errors.join(' ; ')}")
+      raise StandardError.new("Encoding failed for #{source_path}: #{encode.errors.join(' ; ')}")
```

Comment:
I wonder if this is a good opportunity to create our own subclasses of `StandardError` (or, maybe `RuntimeError`) to carry this information consistently and in a machine-readable way?